Index: strongswan-5.0.1/src/libcharon/Makefile.am
===================================================================
--- strongswan-5.0.1.orig/src/libcharon/Makefile.am
+++ strongswan-5.0.1/src/libcharon/Makefile.am
@@ -137,7 +137,7 @@ AM_CFLAGS = \
 	-DIPSEC_DIR=\"${ipsecdir}\" \
 	-DIPSEC_PIDDIR=\"${piddir}\"
 
-libcharon_la_LIBADD = -lm $(PTHREADLIB) $(DLLIB) $(SOCKLIB)
+libcharon_la_LIBADD = -lm $(PTHREADLIB) $(DLLIB) $(SOCKLIB) $(top_builddir)/src/libhydra/libhydra.la $(top_builddir)/src/libstrongswan/libstrongswan.la
 
 EXTRA_DIST = Android.mk
 
Index: strongswan-5.0.1/src/libhydra/Makefile.am
===================================================================
--- strongswan-5.0.1.orig/src/libhydra/Makefile.am
+++ strongswan-5.0.1/src/libhydra/Makefile.am
@@ -11,7 +11,7 @@ kernel/kernel_ipsec.c kernel/kernel_ipse
 kernel/kernel_net.c kernel/kernel_net.h \
 kernel/kernel_listener.h
 
-libhydra_la_LIBADD =
+libhydra_la_LIBADD = $(top_builddir)/src/libstrongswan/libstrongswan.la
 
 INCLUDES = -I$(top_srcdir)/src/libstrongswan
 AM_CFLAGS = \
