diff --git a/conf/plugins/openssl.conf b/conf/plugins/openssl.conf
index fd4658b..7b5d796 100644
--- a/conf/plugins/openssl.conf
+++ b/conf/plugins/openssl.conf
@@ -4,7 +4,7 @@ openssl {
     # engine_id = pkcs11
 
     # Set OpenSSL FIPS mode: disabled(0), enabled(1), Suite B enabled(2).
-    # fips_mode = 0
+    fips_mode = 0
 
     # Whether to load the plugin. Can also be an integer to increase the
     # priority of this plugin.
